### PR TITLE
CORE-6073 NoClassDefFoundError SpecificRecordBase

### PR DESCRIPTION
**Pull request by mbrkic-r3**
- Imported package org.apache.avro.specific for crypto-service-impl

### CODE REVIEW

**Review by chrisr3 — CHANGES_REQUESTED** [review]
Bnd computes the `Import-Package` statement by analysing the byte-code, so how come it misses the `org.apache.avro.specific.*` packages? It's not as if this bundle is rewriting its own byte-code or loading Avro classes dynamically - is it?

On a technical level, the `bnd` elements should be inside a `bundle` section for recent versions of Bnd:
```gradle
tasks.named('jar', Jar) {
    bundle {
        bnd '... instructions...'
    }
}
```

**Comment by mbrkic-r3** [comment]
> Bnd computes the `Import-Package` statement by analysing the byte-code, so how come it misses the `org.apache.avro.specific.*` packages? It's not as if this bundle is rewriting its own byte-code or loading Avro classes dynamically - is it?
> 
> On a technical level, the `bnd` elements should be inside a `bundle` section for recent versions of Bnd:
> 
> ```groovy-gradle
> tasks.named('jar', Jar) {
>     bundle {
>         bnd '... instructions...'
>     }
> }
> ```

https://r3-cev.atlassian.net/browse/CORE-6073?focusedCommentId=204524
WireProcessor.kt is dynamically loading classes that are not imported by crypto-service-impl bundle.
`java.lang.NoClassDefFoundError: org/apache/avro/specific/SpecificRecordBase
	at java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:3137) ~[?:?]
	at java.lang.Class.getConstructors(Class.java:1943) ~[?:?]
	at net.corda.crypto.service.impl.WireProcessor.getHandler$lambda-2(WireProcessor.kt:34) ~[?:?]`

File: components/crypto/crypto-service-impl/build.gradle [hunk 1]
```diff
@@ -45,3 +45,9 @@ dependencies {
 
     testRuntimeOnly "org.slf4j:slf4j-simple:$slf4jVersion"
 }
+
+tasks.named('jar', Jar) {
+    bundle {
+        bnd 'Import-Package: org.apache.avro.specific,*'
```
Comment (chrisr3) [inline components/crypto/crypto-service-impl/build.gradle]: I'll leave the _"What on Earth is `WireProcessor`?"_ question to one side, and just say that it's probably better to use
```
DynamicImport-Package: org.apache.avro.specific
```
for this case, and leave Bnd to manage the `Import-Package` header.

**Review by chrisr3 — CHANGES_REQUESTED** [review]
I think we are using `DynamicImport-Package` for imports that bundles "magically" discover at runtime.

File: components/crypto/crypto-service-impl/build.gradle (hunk 1 above)
Comment (mbrkic-r3) [reply components/crypto/crypto-service-impl/build.gradle]: changed

File: components/crypto/crypto-service-impl/build.gradle [hunk 2]
```diff
@@ -45,3 +45,9 @@ dependencies {
 
     testRuntimeOnly "org.slf4j:slf4j-simple:$slf4jVersion"
 }
+
+tasks.named('jar', Jar) {
+    bundle {
+        bnd 'DynamicImport-Package: org.apache.avro.specific,*'
```
Comment (chrisr3) [inline components/crypto/crypto-service-impl/build.gradle]: Can you check that the `DynamicImport-Package` header written into `META-INF/MANIFEST.MF` _does not_ have `,*` at the end please?

Comment (mbrkic-r3) [reply components/crypto/crypto-service-impl/build.gradle]: Removed ",*" from the end

**Review by chrisr3 — APPROVED** [review]
If this works then its OK by me.

**Pull request merged**